src/main/java/org/radarcns/phone/PhoneLogManager.java

```diff
@@ -240,11 +255,19 @@ public void processSMS(double eventTimestamp, String target, int typeCode, Strin
     /**
      * Extracts last 9 characters and hashes the result with a salt.
      * For phone numbers this means that the area code is removed
-     * E.g.:+31232014111 becomes 232014111 and 0612345678 becomes 612345678 (before hashing)
+     * E.g.: +31232014111 becomes 232014111 and 0612345678 becomes 612345678 (before hashing)
+     * If target is a name instead of a number (e.g. when sms), then hash this name
      * @param target String
      * @return String
      */
     public byte[] createTargetHashKey(String target) {
+        // If non-numerical, then hash the target directly
+        try {
+            Integer.valueOf(target);
```

Comment:
`Long.parseLong()` should be preferred to be able to parse long international numbers (i.e. with a Dutch country code prefix `+31` plus nine digits, the number will exceed `Integer.MAX_VALUE`).